lib/cli/src/commands/package/build.rs

```diff
@@ -0,0 +1,131 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+
+/// Build a container from a package manifest.
+#[derive(clap::Parser, Debug)]
+pub struct CmdPackageBuild {
+    /// Path of the package or wasmer.toml manifest.
+    ///
+    /// Defaults to current directory.
+    #[clap(short = 'p', long)]
+    package: Option<PathBuf>,
+
+    /// Output path for the package file.
+    /// Defaults to current directory + [name]-[version].webc.
+    #[clap(short = 'o', long)]
+    out: Option<PathBuf>,
+}
+
+impl CmdPackageBuild {
+    pub(crate) fn execute(&self) -> Result<(), anyhow::Error> {
+        let manifest_path = if let Some(p) = &self.package {
+            if p.is_dir() {
+                let manifest_path = p.join("wasmer.toml");
+                if !manifest_path.is_file() {
+                    anyhow::bail!(
+                        "Specified directory '{}' does not contain a wasmer.toml manifest",
+                        p.display()
+                    );
+                }
+                manifest_path
+            } else if p.is_file() {
+                p.clone()
+            } else {
+                anyhow::bail!(
+                    "Specified path '{}' is not a file or directory",
+                    p.display()
+                );
+            }
+        } else {
+            let dir = std::env::current_dir().context("could not get current directory")?;
+            let manifest_path = dir.join("wasmer.toml");
+            if !manifest_path.is_file() {
+                anyhow::bail!(
+                    "Current directory '{}' does not contain a wasmer.toml manifest - specify a path with --package-dir",
+                    dir.display()
+                );
+            }
+            manifest_path
+        };
+
+        let pkg = webc::wasmer_package::Package::from_manifest(manifest_path)?;
+
+        let manifest = pkg
+            .manifest()
+            .package_annotation::<webc::metadata::annotations::Wapm>("wapm")
```

Comment:
Prefer to use the helper methods where possible. That way we don't use magic strings (although you also have access to `webc::metadata:annotations::Wapm::KEY`) and there's no possibility of deserializing to the wrong type.

```suggestion
            .wapm()
```